### PR TITLE
 Finishing : Set up the Front-End

### PR DESCRIPTION
**Pull request by jbrandy1**
I was struggling to figure out why reloading the browser didn't allow changes to happen in the UI. I'm not sure if I did something wrong when creating the dist directory because the page source of the  http://127.0.0.1:8000 link is different from the index.html inside of the UI folder. 

Here is what my current screen is showing me
Screenshot_24-4-2024_131327_127 0 0 1

When my jsx file looks like this:
Screenshot 2024-04-24 131442

Plus I'm not sure if my route root changed at all according to these screenshots I took
Screenshot_24-4-2024_13116_127 0 0 1

### CODE REVIEW

**Comment by jbrandy1** [comment]
Any advice as to what I should check for would be extremely helpful @semmypurewal. Thank you!

File: ui/node_modules/.bin/esbuild
Comment (semmypurewal) [inline ui/node_modules/.bin/esbuild:1]: You should really add `node_modules` to your `.gitignore` file so there's not a lot of extraneous stuff here.

**Review by semmypurewal — APPROVED** [review]
This looks good, so I'll approve it for now. Let me take a look at your question in the main comments.

**Comment by semmypurewal** [comment]
> Any advice as to what I should check for would be extremely helpful @semmypurewal. Thank you!

I'm not exactly sure I understand the issue. Is the program not rebuilding when you make a change to the JavaScript or to the Fast API server? From the question, it looks like the JavaScript side. But I don't see your docker compose or anything as part of this change, so I'm not sure what I'm looking for.

Or maybe you already have a docker compose file from a prior PR?

**Comment by semmypurewal** [comment]
Yeah, I don't think this will work if you don't have a dockerfile or a docker compose file set up, and I don't see one in the repository.

**Review by jbrandy1 — COMMENTED** [review]
@semmypurewal  I accidentally made the third pull request before I merged the second pull request, so it's also tagged in this one. An issue I was going to address in the third pull was that changes made in the main.jsx file for the UI still didn't show up, so I'm still having some UI issues

**Pull request merged**